src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -2292,6 +2364,60 @@ Vector2i WindowImplX11::getPrimaryMonitorPosition()
     return monitorPosition;
 }
 
+void WindowImplX11::setItemDroppingEnabled(bool enabled)
+{
+    // In order for file dropping to be enabled, the XdndAware property must be set.
+    if (enabled)
+    {
+        Atom xdnd_version = 5;
+        XChangeProperty(m_display,
+                        m_window,
+                        getAtom("XdndAware"),
+                        XA_ATOM,
+                        32,
+                        PropModeReplace,
+                        reinterpret_cast<unsigned char*>(&xdnd_version),
+                        true);
+    }
+    else
+    {
+        XDeleteProperty(m_display, m_window, getAtom("XdndAware"));
+    }
+}
+
+sf::String WindowImplX11::getWindowStringProperty(Display* dpy, Window w, Atom p)
+{
+    // Taken from https://www.uninformativ.de/blog/postings/2017-04-02/0/POSTING-en.html
+
+    Atom           da, incr, type;
+    int            di;
+    unsigned long  size, dul;
+    unsigned char* prop_ret = NULL;
+
+    // Dummy call to get type and size.
+    XGetWindowProperty(dpy, w, p, 0, 0, false, AnyPropertyType, &type, &di, &dul, &size, &prop_ret);
+    XFree(prop_ret);
+
+    incr = XInternAtom(dpy, "INCR", False);
+    if (type == incr)
+    {
+        err() << "Data too large and INCR mechanism not implemented\n";
+        return "";
+    }
+
+    // Read the data in one go.
+    XGetWindowProperty(dpy, w, p, 0, static_cast<long>(size), false, AnyPropertyType, &da, &di, &dul, &dul, &prop_ret);
+
+    String value = reinterpret_cast<char*>(prop_ret);
+
+    XFree(prop_ret);
+
+    // Signal the selection owner that we have successfully read the data.
+    XDeleteProperty(dpy, w, p);
```

Comment:
We have two `XGetWindowProperty` calls, but only one `XDeleteProperty`.

Is this correct? Do we not leak memory?

Reply:
see https://github.com/SFML/SFML/pull/2265/files/f49594a978e8c1307ace93a505c9a49cbf37433f#r1026700494